rust/yarp/src/lib.rs

```diff
@@ -0,0 +1,312 @@
+//! # yarp
+//!
+//! Rustified version of Ruby's YARP parser.
+//!
+#![warn(
+    box_pointers,
```

Comment:
I see clippy is warning about box-pointer use, but we probably want to allow that, and thus this line could go away.